src/unitxt/processors.py

```diff
@@ -277,3 +277,29 @@ def process_value(self, text: Any) -> Any:
 
         except:
             return 0
+
+
+class ExtractMtBenchStringJudgment(FieldOperator):
```

Comment:
Why is this called ExtractMtBenchStringJudgment ? It's not an MtBench format, if I understand right.  You basically want to extract the first word, and then map "yes" to 1.0 , "no" to 0, and all other values to 0.5.

This can mostly be done with existing post_processors as in here:

prepare/templates/classification/grammatical_error_detection.py

```
add_to_catalog(
    InputOutputTemplate(
        instruction="""You are given a text. Does this text contain any grammatical errors or spelling mistakes? Answer only "Yes" or "No".\n""",
        input_format="Text: {text}",
        output_format="{label}",
        target_prefix="Answer: ",
        postprocessors=[
            "processors.take_first_word",
            "processors.lower_case",
            "processors.yes_no_to_int",
            "processors.cast_to_float_return_nan_if_failed",
        ],
    ),
    "templates.grammatical_error_detection.yes_no",
    overwrite=True,
)
```

Thus you could more generic and simpler post processor:

add_to_catalog(
    SequentialOperator(
        steps=[
            CastFields(
                fields={"prediction": "float"},
                failure_defaults={"prediction": 0.5},
            ),
            CastFields(
                fields={"references": "float"},
                process_every_value=True,
            ),
        ]
    ),
    "processors.cast_to_float_return_0_5_if_failed",
    overwrite=True,
)

Reply:
Yes, you are right. I have now defined this post-processor in the test script `evaluate_ensemble_judge.py` I am happy to change the location where you deem this fit.